TLM/TLM/UI/MainMenu/MainMenuPanel.cs

```diff
@@ -206,5 +207,37 @@ public class SizeProfile {
 			absolutePosition = rect.position;
 			Invalidate();
 		}
-	}
+
+                public void OnGUI() {
+                        // Some safety checks to not trigger while full screen/modals are open
+                        // Check the key and then click the corresponding button
+                        if (!UIView.HasModalInput()
+                            && !UIView.HasInputFocus()) {
+                                if (OptionsKeymapping.KeyToggleTrafficLightTool.IsPressed(Event.current)) {
```

Comment:
I think that checking if key is pressed while TM:PE menu isn't currently open seems to be pointless.

Reply:
I believe that OnGUI will not run for an element if it is not visible.
I do not understand what should be changed here.

Reply:
I've just checked what is going on when we hiding panel and:
Unity docs: `If the MonoBehaviour's enabled property is set to false, OnGUI() will not be called`
but...
In this case `OnGUI` is called, because our `MainMenuPanel: UIPanel` is not `disabled` but `hidden` using internal flag `UIComponent.isVisible`

I suggest adding `this.isVisible` just before `!UIView.HasModalInput()` check and I think we can merge because everything is working right.

@aubergine10 what do you think?